File: engine/model/src/main/scala/net/server/channel/packet/pet/PetMovementPacket.scala
package net.server.channel.packet.pet

import net.server.MaplePacket
import server.movement.LifeMovementFragment

class PetMovementPacket(private var _petId: Int, private var _numberOfCommands: Byte, private var _commands: java.util.List[LifeMovementFragment]) extends MaplePacket {
  def petId: Int = _petId

  def numberOfCommands: Byte = _numberOfCommands

  def commands: java.util.List[LifeMovementFragment] = _commands
}

File: engine/engine-model/src/main/java/net/server/channel/packet/pet/PetMovementPacket.java
package net.server.channel.packet.pet;

import java.util.List;

import net.server.MaplePacket;
import server.movement.LifeMovementFragment;

public record PetMovementPacket(int petId, byte numberOfCommands,
                                List<LifeMovementFragment> commands) implements MaplePacket {
}
